comp/dogstatsd/server/server.go

```diff
@@ -187,19 +189,20 @@ func initTelemetry(cfg config.Reader, logger logComponent.Component) {
 	packets.InitTelemetry(get("telemetry.dogstatsd.listeners_channel_latency_buckets"))
 }
 
-// TODO: (components) - remove once serverless is an FX app
-//
-//nolint:revive // TODO(AML) Fix revive linter
-func NewServerlessServer() Component {
-	return newServerCompat(config.Datadog, logComponentImpl.NewTemporaryLoggerWithoutInit(), replay.NewServerlessTrafficCapture(), serverdebugimpl.NewServerlessServerDebug(), true)
-}
-
 // TODO: (components) - merge with newServerCompat once NewServerlessServer is removed
 func newServer(deps dependencies) Component {
-	return newServerCompat(deps.Config, deps.Log, deps.Replay, deps.Debug, deps.Params.Serverless)
+	s := newServerCompat(deps.Config, deps.Log, deps.Replay, deps.Debug, deps.Params.Serverless, deps.Demultiplexer)
+
+	deps.Lc.Append(fx.Hook{
+		OnStart: s.start,
```

Comment:
[Here](https://github.com/DataDog/datadog-agent/pull/22584/files#diff-539d28e7c20e0a4fa8fb915e53c4b3733edd7f53e1342192207aafb9b338cd99L598) when `start` returns an error the application logs the error whereas with your changes the application stops. Is this behavior expected?

Reply:
I changed the logic a bit here now so that when the server is running, it will log similarly instead. Good Catch! 🙇🏼‍♂️